### PR TITLE
Feature/add person student teacher class

### PR DESCRIPTION
**Pull request by shahadat3669**
### Project requirements
- Create a class Person with the following:
  - Instance vars: @id, @name, and @age.
  - Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.
  - Getters for @id, @name, and @age.
  - Setters for @name and @age.
  - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
  - Public method can_use_services? that returns true if a person is of age or if they have permission from parents.
- Create class Student with the following:
  - Inherits from Person.
  - Constructor extends parent's constructor by adding @classroom and a parameter for it.
  - Method play_hooky that returns "¯\(ツ)/¯".
- Create a class Teacher with the following:
  - Inherits from Person.
  - Constructor extends parent's constructor by adding @specialization and a parameter for it.
  - Override can_use_services? so it always returns true.
  - Each class should be saved in a separate file.

### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correctly [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (OyePriscilla) [inline person.rb]: - [x] Kindly the make this method private according to the project requirement. Private method can ONLY be called from inside the class where it's defined; this allows you to control access to your methods. ℹ️

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class Teacher < Person
+  def initialize(name = 'Unknown', age = nil, parent_permission: true, specialization: nil)
+    super(name, age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    super || @specialization
+  end
```
Comment (OyePriscilla) [inline teacher.rb]: - [x] I think this method `can_use_services?` should always return true according to the project requirement.
```suggestion
  def can_use_services?
    true
  end
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,178 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
+
+**Shahadat Hossain**
+
+[![portfolio](https://img.shields.io/badge/my_portfolio-000?style=for-the-badge&logo=ko-fi&logoColor=white)](https://github.com/shahadat3669) [![linkedin](https://img.shields.io/badge/shahadat3669-0A66C2?style=for-the-badge&logo=linkedin&logoColor=white)](https://linkedin.com/in/shahadat3669) [![twitter](https://img.shields.io/badge/@shahadat3669-1DA1F2?style=for-the-badge&logo=twitter&logoColor=white)](https://twitter.com/shahadat3669)
+
```
Comment (OyePriscilla) [inline README.md:123]: - [optional] Please kindly format this section of your README file neatly, README file most times has the capability to attract developers/visitors to our project. 👏

File: README.md [hunk 4]
```diff
@@ -1,2 +1,178 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- CONTRIBUTING -->
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
```
Comment (OyePriscilla) [inline README.md]: - [x] Please kindly fix this broken link to make your work look professional. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @shahadat3669 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implement person, student and teacher classes ✔️ 
Implement inheritance ✔️ 
Use Ruby Getters and Setters method ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @shahadat3669 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Make all changes requested by the previous reviewer
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**